python/pyarrow/__lib_pxi/builder.pyi

```diff
@@ -0,0 +1,106 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from typing import Iterable
+
+from pyarrow.lib import MemoryPool, _Weakrefable
+
+from .array import StringArray, StringViewArray
+
+class StringBuilder(_Weakrefable):
+    """
```

Comment:
Why are docstrings duplicated in the stubs files? Do they need to be kept in sync? Why (and how)?

Reply:
See my reply [here](https://github.com/apache/arrow/discussions/45919#discussioncomment-14264835). I am not yet sure we want to include docstrings here, but if we do we should definitely script their extraction. `stubgen --include-docstrings` does this but I suspect it doesn't read cython-only symbols.